src/test/java/seedu/address/logic/commands/BuyCommandTest.java

```diff
@@ -330,6 +336,16 @@ public void addTransaction(Transaction transaction) {
         public void updateFilteredTransactionList(Predicate<Transaction> predicate) {
             throw new AssertionError("This method should not be called.");
         }
+
+        @Override
+        public void commit() {
+            return;
```

Comment:
Could you put "throw new AssertionError("This method should not be called.");" for the commit() body as well? That way we will have an informative error if the method is accidentally called.

Reply:
BuyCommand and SellCommand will call undo since they are modifying, so if I throw Exception here, it will fail.

I'm still not sure how to isolate the commit call

Reply:
I see you have made the ModelStubCommit in it's own java file. I think that works too.

Reply:
I don't want to make the same class for every command test...